job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/StepContext.groovy

```diff
@@ -42,6 +42,27 @@ class StepContext extends AbstractExtensibleContext {
     }
 
     /**
+     * Runs a xshell script.
+     *
+     * @since 1.41
+     */
+    @RequiresPlugin(id = 'xshell', minimumVersion = '0.10')
+    void xshell(@DslContext(XShellContext) Closure xShellClosure) {
+        XShellContext xShellContext = new XShellContext(jobManagement)
+        ContextHelper.executeInContext(xShellClosure, xShellContext)
+        stepNodes << new NodeBuilder().'hudson.plugins.xshell.XShellBuilder' {
+            delegate.executableInWorkspaceDir(xShellContext.executableInWorkspaceDir)
```

Comment:
Awesome! I did get confused when I did replace those matches, thanks for that spot! I've already changed, squashed and pushed
Thanks